src/util-config.h

```diff
@@ -40,6 +42,9 @@ enum ConfigType {
     CONFIG_TYPE_FILE,       /* file logging */
     CONFIG_TYPE_PCAP,       /* pcap logging */
     CONFIG_TYPE_DROP,       /* drop logging */
+    CONFIG_TYPE_REASSEMBLY, /**< stream reassembly */
+    CONFIG_TYPE_APP_PARSER, /**< app parsing */
+    CONFIG_TYPE_APP_FILES,  /**< app files */
```

Comment:
1. Lots of TBD types here. Should they be marked?
2. Should the scope be made a part of type like it is for `APP` types?